modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/A11y.tsx

```diff
@@ -12,185 +12,214 @@
+import useIframeA11yChannel from './hooks/useIframeA11yChannel';
+import {Kind} from './hooks/useIframeClient';
 
 import type {A11yCheckerOptions} from './A11yChecker';
-import type {Violation as TViolation} from './useA11y';
-
-const rectAttrs: Array<keyof DOMRect> = [
-	'bottom',
-	'height',
-	'left',
-	'right',
-	'top',
-	'width',
-];
-
-interface IRectState {
-	rect: DOMRect | undefined;
-	hasRectChanged: boolean;
-}
 
-const DOMRectStub = {} as DOMRect;
+type Params = {
+	name?: string;
+	ruleId?: string;
+	target?: string;
+};
 
-const rectChanged = (a: DOMRect, b: DOMRect = DOMRectStub) =>
-	rectAttrs.some((prop) => a[prop] !== b[prop]);
+type IframePayload = {
+	ruleId: string;
+	target: string;
+};
 
-const useObserveRect = (
-	callback: (rect: DOMRect | undefined) => void,
-	node: Element | null
-) => {
-	const rafIdRef = useRef<number>();
+export function A11y(props: Omit<A11yCheckerOptions, 'callback'>) {
+	const violations = useA11y(props);
 
-	const run = useCallback(
-		(node: Element, state: IRectState) => {
-			const newRect = node.getBoundingClientRect();
+	const [state, dispatch] = useFilterViolations(violations);
 
-			if (rectChanged(newRect, state.rect)) {
-				state.rect = newRect;
+	const [params, setParams] = useState<Params>();
 
-				callback(state.rect);
-			}
+	const [activePage, setActivePage] = useState(0);
 
-			rafIdRef.current = window.requestAnimationFrame(() =>
-				run(node, state)
-			);
-		},
-		[callback]
-	);
+	const nodes = Object.keys(violations.nodes);
 
-	useEffect(() => {
-		if (node) {
-			run(node, {
-				hasRectChanged: false,
-				rect: undefined,
-			});
+	useLayoutEffect(() => {
+		if (nodes.length > 0) {
+			document.body.classList.add('a11y-body');
 
 			return () => {
-				if (rafIdRef.current) {
-					cancelAnimationFrame(rafIdRef.current);
-				}
+				document.body.classList.remove('a11y-body');
 			};
 		}
-	}, [node, run]);
-};
-
-const Overlay = React.forwardRef<
-	HTMLDivElement,
-	React.ButtonHTMLAttributes<HTMLDivElement>
->(({style, ...othersProps}, ref) => (
-	<ReactPortal
-		{...othersProps}
-		className="a11y-overlay"
-		ref={ref}
-		style={style}
-	>
-		<div className="a11y-indicator">
-			<ClayIcon symbol="info-circle" />
-		</div>
-		<div className="a11y-backdrop" />
-	</ReactPortal>
-));
-
-type ViolationProps = {
-	modifyIndex: number;
-	target: string;
-	violations: Array<TViolation>;
-};
+	}, [nodes]);
 
-function Violation({target, violations}: ViolationProps) {
-	const [visible, setVisible] = useState(false);
-	const [bounds, setBounds] = useState<React.CSSProperties>();
+	const navigateToOccurrence = useCallback(
+		(ruleId: string, target: string) => {
+			dispatch({
+				payload: {key: 'id', value: ruleId},
+				type: TYPES.ADD_FILTER,
+			});
+			dispatch({
+				payload: {key: 'nodes', value: target},
+				type: TYPES.ADD_FILTER,
+			});
+			setActivePage(2);
+			setParams({
+				name: Liferay.Util.sub(
+					Liferay.Language.get('occurrence-x'),
+					'1'
```

Comment:
With this `1` here, Are we just navigating to the first occurrence? I think it cannot be hard coded

Reply:
We don't actually navigate to the first occurrence, we don't use the index order to say which occurrence should be rendered but we use the corresponding ids, this is just for the name to be rendered in the panel title, so it wouldn't be a problem, at this level we don't know what is the position of the occurrence to add here.

Because we are interacting at the Overlay level, to get the `index` that the Panel is rendering we would have to change a lot for that, which is not necessary. For example, the Panel at the occurrences component level will convert the object to an Array but only this component knows what the position is, in this case, we would have to do the same and get a `findIndex` to get the index, it would be good to avoid doing this here , because it is an unnecessary cost.